src/common/user.ts

```diff
@@ -47,16 +47,20 @@ export const deleteUser = async (
       }
 
       if (subscriptionFlags?.provider === SubscriptionProvider.Paddle) {
-        await cancelSubscription({
-          subscriptionId: subscriptionFlags.subscriptionId,
-        });
+        const isGifted = !!subscriptionFlags.giftExpirationDate;
+        // gifted subscription is a one-time payment hence not considered subscription in Paddle's terms
+        if (!isGifted) {
+          await cancelSubscription({
```

Comment:
Can we add a test for this case and validate `cancelSubscription` is not called?

Reply:
Good point!